hls4ml/backends/vivado/passes/conv_same_pad.py

```diff
@@ -7,7 +7,7 @@ class InsertZeroPaddingBeforeConv1D(OptimizerPass):
 
     def match(self, node):
         is_match = (
-            isinstance(node, (Conv1D, SeparableConv1D))
+            isinstance(node, (Conv1D, SeparableConv1D, DepthwiseConv1D))
             and ((node.get_attr('padding') == 'same') or (node.get_attr('padding') == 'causal'))
```

Comment:
I think there needs to be a better check than if the padding is "same" or "causal".

Reply:
@jmitrevs what is best in your opinion, searching for an equivalent condition to check if padding is `same` or `causal`, or setting at conversion time the attribute `padding` to `same` or `causal` (as it happens for the Keras models) using the same aforementioned condition? I'd opt for the second one

Reply:
In onnx arent't the padding sizes explicit

Reply:
I am proposing this change in general, which should make these changes not needed:
https://github.com/fastmachinelearning/hls4ml/pull/1061

Reply:
I think `conv_same_pad.py` should just be updated to the version currently in the qonnx_1p0 branch. (I think that might be the source of the merge conflicts.)